src/modalities/checkpointing/checkpoint_loading.py

```diff
@@ -4,8 +4,30 @@
 import torch.nn as nn
 from torch.optim import Optimizer
 
+from modalities.checkpointing.stateful.app_state import AppState
 
-class CheckpointLoadingIF(ABC):
+
+class DistributedCheckpointLoadingIF(ABC):
+    """Distributed checkpoint loading interface for loading PyTorch models and optimizer checkpoints."""
+
+    @abstractmethod
+    def load_checkpoint_(self, app_state: AppState, checkpoint_directory_path: Path) -> AppState:
+        """Loads the distributed checkpoint from the specified directory path into the AppState.
+
+        Args:
+            app_state (AppState): The application state with the model, optimizer and lr scheduler.
+            checkpoint_directory_path (Path): The directory path to the distributed checkpoint.
+
+        Raises:
+            NotImplementedError: This abstract method is not implemented and should be overridden in a subclass.
+
+        Returns:
+            AppState: The application state with the loaded checkpoint.
+        """
+        raise NotImplementedError
+
+
+class FSDP1CheckpointLoadingIF(ABC):
```

Comment:
Does the interface contain FSDP1-specific logic for loading checkpoints? If not, I would opt keeping the name "CheckpointLoadingIF" or "NoneDistributedCheckpointLoadingIF".

Reply:
With the current setup, FSDP2 always requires DCP and with FSDP1 we resort to the old checkpointing implementation (as before). 
Since we will throw out FSDP1 eventually, I renamed this IF to be FSDP1-specific (so that FSDP2 is never used with this checkpointing implementation).